com.unity.render-pipelines.high-definition/Runtime/Material/Decal/Decal.hlsl

```diff
@@ -81,7 +81,11 @@ void DecodeFromDBuffer(
     // Use (254.0 / 255.0) instead of 0.5 to allow to encode 0 perfectly (encode as 127)
     // Range goes from -0.99607 to 1.0039
     surfaceData.normalWS.xyz = inDBuffer1.xyz * 2.0 - (254.0 / 255.0);
-    surfaceData.normalWS.w = inDBuffer1.w;
+
+    // Case 1317162: Apply a NaN guard via the normal blend mask.
+    // An epsilon will prevent a situation where two opposing normals (surface & decal) produce a 0 normal from blending.
+    surfaceData.normalWS.w = inDBuffer1.w + 1e-5;
```

Comment:
I am concern by this fix. I am not confortable with touching the alpha value as we will be unable to know if we are pure 0 or not.  Maybe we should do surfaceData.normalWS.w = inDBuffer1.w > 0 ? saturate(inDBuffer1.w + 1e-5) : 0;
thought?

Reply:
ah currently guess it is the opposite and need to check pure 1

Reply:
As discussed in slack we will replace this approach by instead safe normalizing at all occurrences of `ApplyDecalToSurfaceData`. Note that in the future we will introduce surface gradient framework to decals which will fix this implicitly and remove the need for safe normalize.